### PR TITLE
Obsolete MigrationsEndpointMiddleware

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Following up our offline discussion where we decided to remove MigrationsEndpointMiddleware. Note that I added the RazorPageGenerator (originally from aspnetcore-tooling) since we need that to generate the Designer.cs file.

### CODE REVIEW

File: src/Middleware/Diagnostics.EntityFrameworkCore/src/DatabaseErrorPageMiddleware.cs [hunk 1]
```diff
@@ -59,7 +58,9 @@ public void Hold(Exception exception, Type contextType)
         public DatabaseErrorPageMiddleware(
```
Comment (Tratcher) [inline src/Middleware/Diagnostics.EntityFrameworkCore/src/DatabaseErrorPageMiddleware.cs:58]: Make a new overload that doesn't have the obsolete parameter.

File: src/Middleware/Diagnostics.EntityFrameworkCore/src/Views/DatabaseErrorPageModel.cs [hunk 2]
```diff
@@ -9,6 +9,21 @@ namespace Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Views
...
+            IEnumerable<string> pendingMigrations)
+        {
+            ContextType = contextType;
+            Exception = exception;
+            DatabaseExists = databaseExists;
+            PendingModelChanges = pendingModelChanges;
+            PendingMigrations = pendingMigrations;
+        }
+
+        [Obsolete("This is obsolete and will be removed in a future version.")]
```
Comment (Tratcher) [inline src/Middleware/Diagnostics.EntityFrameworkCore/src/Views/DatabaseErrorPageModel.cs:26]: This class is internal, you can just remove this overload and the property.

File: src/Middleware/Diagnostics.EntityFrameworkCore/test/UnitTests/DatabaseErrorPageOptionsTest.cs [hunk 3]
```diff
@@ -11,18 +12,22 @@ public class DatabaseErrorPageOptionsTest
         [Fact]
         public void Empty_MigrationsEndPointPath_by_default()
         {
+#pragma warning disable CS0618 // Type or member is obsolete
             var options = new DatabaseErrorPageOptions();
```
Comment (Tratcher) [inline src/Middleware/Diagnostics.EntityFrameworkCore/test/UnitTests/DatabaseErrorPageOptionsTest.cs:16]: Remove this whole test class.

File: src/Middleware/Diagnostics.EntityFrameworkCore/src/MigrationsEndPointMiddleware.cs [hunk 4]
```diff
@@ -15,6 +15,7 @@ namespace Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore
     /// <summary>
     /// Processes requests to execute migrations operations. The middleware will listen for requests to the path configured in the supplied options.
     /// </summary>
+    [Obsolete("This is obsolete and will be removed in a future version. Use the Package Manager Console in Visual Studio or dotnet-ef tool on the command line to apply migrations.", error:true)]
     public class MigrationsEndPointMiddleware
```
Comment (Tratcher) [inline src/Middleware/Diagnostics.EntityFrameworkCore/src/MigrationsEndPointMiddleware.cs:19]: This is dead code now. Remove all internal content and make the public APIs throw NotImplementedException.

**Review by Tratcher — CHANGES_REQUESTED** [review]

**Review by Tratcher — APPROVED** [review]

**Comment by JunTaoLuo** [comment]
Closing this since we are taking a different approach to this issue.

**Pull request closed without merging**